scripts/generate_presign_url/main.py

```diff
@@ -2,69 +2,106 @@
-    parser.add_argument('-o', '--out-file',
-                        default='presigned-urls.txt',
-                        help="Name of the output file. Default: presigned-urls.txt")
-    parser.add_argument('--dryrun',
-                        default=False,
-                        action='store_true',
-                        help="Perform a dry run.")
-    parser.add_argument('-t', '--filetype',
-                        default=['FASTQ', 'BAM', 'CRAM', 'VCF'],
-                        nargs='+',
-                        choices=['FASTQ', 'BAM', 'CRAM', 'VCF'],
-                        help='Filetype to filter by (FASTQ, BAM, CRAM,VCF). Space separated if more than one. Default: all files')
-    parser.add_argument('-s', '--study-ids',
-                        required=True,
-                        nargs='+',
-                        help="AGHA study ID(s) to retrieve files for. Space separated if more than one.")
-    flagship_list = list(set(agha.FlagShip.list_flagship_enum()) - {'UNKNOWN', 'TEST'})
-    parser.add_argument('-f', '--flagship',
-                        required=True,
-                        choices=flagship_list,
-                        help="Code of the flagship the sample belongs to.")
+    parser = argparse.ArgumentParser(
+        description="Generate pre-signed URLs for AGHA files."
+    )
+    parser.add_argument(
+        "-o",
+        "--out-file",
+        default="presigned-urls.txt",
+        help="Name of the output file. Default: presigned-urls.txt",
+    )
+    parser.add_argument(
+        "--dryrun", default=False, action="store_true", help="Perform a dry run."
+    )
+    parser.add_argument(
+        "-t",
+        "--filetype",
+        default=["FASTQ", "BAM", "CRAM", "VCF"],
+        nargs="+",
+        choices=["FASTQ", "BAM", "CRAM", "VCF"],
+        help="Filetype to filter by (FASTQ, BAM, CRAM,VCF). Space separated if more than one. Default: all files",
+    )
+    parser.add_argument(
+        "-s",
+        "--study-ids",
+        required=True,
+        nargs="+",
+        help="AGHA study ID(s) to retrieve files for. Space separated if more than one.",
+    )
+    parser.add_argument(
+        "--release-id",
+        required=True,
+        help="A unique ID to identify the sharing request. "
+        "This ID will be inserted as a custom parameter in the presignedUrl for access tracking.",
+    )
+    flagship_list = list(set(agha.FlagShip.list_flagship_enum()) - {"UNKNOWN", "TEST"})
+    parser.add_argument(
+        "-f",
+        "--flagship",
+        required=True,
+        choices=flagship_list,
+        help="Code of the flagship the sample belongs to.",
+    )
     args = parser.parse_args()
 
     print("######################" * 6)
-    print('Running the following')
+    print("Running the following")
     print(f"Filetype : {args.filetype}")
     print(f"Flagship : {args.flagship}")
     print(f"IDs      : {args.study_ids}")
```

Comment:
Do you want to add the release ID here, as it's a mandatory argument and valuable in case you want to record that stdout?

Reply:
Yes, added in!